iceoryx_posh/source/popo/ports/server_port_user.cpp

```diff
@@ -61,20 +71,72 @@ bool ServerPortUser::hasLostRequestsSinceLastCall() noexcept
 }
 
 cxx::expected<ResponseHeader*, AllocationError>
-ServerPortUser::allocateResponse(const uint32_t /*userPayloadSize*/) noexcept
+ServerPortUser::allocateResponse(const RequestHeader* const requestHeader,
+                                 const uint32_t userPayloadSize,
+                                 const uint32_t userPayloadAlignment) noexcept
 {
-    /// @todo
-    return cxx::error<AllocationError>(AllocationError::RUNNING_OUT_OF_CHUNKS);
+    cxx::Ensures(requestHeader != nullptr && "requestHeader must not be a nullptr");
+    if (requestHeader == nullptr)
```

Comment:
Those lines would be gone if request header would be a reference.

Reply:
Like I already mentioned in the PR description, I'm not that happy with the interface. It is modeled after the pub-sub interface where we also use pointer. I think in this case we also cannot easily use references since it make the user interface also error prone. Since this is used in combination with `getRequest`, which returns a pointer, it would be kind of awkward to call `releaseRequest(*req)` so we should also return a reference with `getRequest`. But one can easily write `auto req = requestResult.value()` and create a copy on the stack which results in an error when `releaseRequest` is called. One solution to this problem might be to use a `std::reference_wrapper`. This would fix the problem of the unintentional copy and would guarantee to not be `null`. I do not fully estimate the consequences of this change and if we do it, we should also do it for pub-sub. Therefore I would suggest to keep it as is for the next release and create an issue to replace the raw pointer with `std::reference_wrapper` where appropriate or create an own abstraction if `std::reference_wrapper` does not fulfill our needs.

Reply:
@elfenpiff I created #1071 and will close all your comments regarding references if you agree?